pkg/controller/operators/catalog/operator.go

```diff
@@ -533,20 +595,35 @@ func (o *Operator) syncSubscriptions(obj interface{}) error {
 			break
 		}
 	}
-
 	installplanReference, err := o.createInstallPlan(namespace, subs, installPlanApproval, steps)
 	if err != nil {
 		logger.WithError(err).Debug("error creating installplan")
 		return err
 	}
 
-	if err := o.ensureSubscriptionInstallPlanState(namespace, subs, installplanReference); err != nil {
+	if err := o.updateSubscriptionSetInstallPlanState(namespace, subs, installplanReference); err != nil {
 		logger.WithError(err).Debug("error ensuring subscription installplan state")
 		return err
 	}
 	return nil
 }
 
+func (o *Operator) syncSubscriptions(obj interface{}) error {
+	sub, ok := obj.(*v1alpha1.Subscription)
+	if !ok {
+		o.Log.Debugf("wrong type: %#v", obj)
+		return fmt.Errorf("casting Subscription failed")
+	}
+
+	o.resolveNamespace(sub.GetNamespace())
```

Comment:
Unit test needs to be changed from calling `syncSubscription()` to `syncResolvingNamespace()` since the logic was moved to that method.

Reply:
Just pushed up that change